packages/e2e-test-app/test/XAMLTest.test.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ *
+ * @format
+ */
+
+import {dumpVisualTree} from '@react-native-windows/automation-commands';
+import {goToComponentExample} from './RNTesterNavigation';
+import {app} from '@react-native-windows/automation';
+
+beforeAll(async () => {
+  await goToComponentExample('XAML');
+});
+
+describe('XamlTest', () => {
+  test('Layout', async () => {
+    const dump = await dumpVisualTree('ReactNativeXAMLRoot');
+    expect(dump).toMatchSnapshot();
+  });
+  test('MenuFlyoutIsOpen', async () => {
+    const tappableTextBlock = await app.findElementByTestID(
+      'tappableTextBlock',
+    );
+    await tappableTextBlock.click();
+    const dump = await dumpVisualTree('MenuFlyout');
+    expect(dump).toMatchSnapshot('ReactNativeXAML-MenuFlyout');
+    const menuOption2 = await app.findElementByTestID('menuOption2');
+    await menuOption2.click();
+    const text = await tappableTextBlock.getText();
+    expect(text).toBe('menu option 2 clicked');
```

Comment:
There is a race here. We sent a click via UIA, but the text is updated asynchronously. You can use `app.WaitUntil` to poll for the condition to be met.

Note that the above `click`, etc are fine. WinAppDriver will retry finding an alement until it is available. A first valid value will be returned after performing an action though.

Reply:
good catch.

one thing I'm seeing (unrelated) is that the dump for the menu isn't generated/collected in CI... any idea what's missing?

Reply:
It looks like something is crashing. Reminder for me to figure out why dumps aren't being collected...

Usually when I've seen CI crashes that do not happen locally, it is from using APIs that are unsupported on Server 2019 (Windows 17763). Something to check for here.

Reply:
It's also possible the e2e-test-app output is bad... The error messages we have right now leave a bit to be desired.

I think you might also want to poll in these two lines, to ensure the flyout is open before trying to dump it.

```
await tappableTextBlock.click();
const dump = await dumpVisualTree('MenuFlyout');
```

Reply:
I figured this one out - the menuFlyout doesn't actually end up in the visual tree. XAML will create a MenuFlyoutPresenter (which doesn't carry the automation ID from the MenuFlyout) and put that in a separate visual root separate from the App's. The XAML VisualTreeHelper APIs don't seem to be able to reason over children of popups, so we need some special casing, see latest commit.